lib/FusionInventory/Agent/Task/Inventory/Linux/Networks/DockerMacvlan.pm

```diff
@@ -0,0 +1,90 @@
+package FusionInventory::Agent::Task::Inventory::Linux::Networks::DockerMacvlan;
+
+use strict;
+use warnings;
+
+use parent 'FusionInventory::Agent::Task::Inventory::Module';
+
+use JSON::PP;
+use FusionInventory::Agent::Tools;
+use FusionInventory::Agent::Tools::Network;
+
+sub isEnabled {
+    return canRun('docker');
+}
+
+sub doInventory {
+    my (%params) = @_;
+
+    my $inventory = $params{inventory};
+    my $logger    = $params{logger};
+
+    my @networks = _getMacvlanNetworks(logger => $logger);
+
+    foreach my $network (@networks) {
+        my @interfaces = _getInterfaces(logger => $logger, networkId => $network);
+        foreach my $interface (@interfaces) {
+            $inventory->addEntry(
+                section => 'NETWORKS',
+                entry   => $interface
+            );
+        }
+    }
+}
+
+sub _getMacvlanNetworks {
+    my (%params) = (
+        command => 'docker network ls --filter driver=macvlan -q',
+        @_
+    );
+    my $handle = getFileHandle(%params);
+    return unless $handle;
+
+    my @networks;
+
+    while (my $line = <$handle>) {
+        chomp($line);
+        push @networks, $line;
+    }
+
+    close $handle;
+    return @networks;
+}
+
+sub _getInterfaces {
+    my (%params) = @_;
+
+    $params{command} = "docker network inspect $params{networkId}";
+
+    my $lines = getAllLines(%params);
```

Comment:
an `or return;` is missing here in the case `$lines` is not defined.

Reply:
Will do. Although if $lines is not defined, the iterator a few lines below will just do nothing.

Reply:
Not exactly, the `$json->decode($lines);` will fail. This is safe as you put the code in an eval, but this is better to handle the case before.